### PR TITLE
Implement getVehicleSoundParameters

### PR DESCRIPTION
**Pull request by duncanspumpkin**
This one is probably for after the OpenAL PR as its hard to verify without being able to hear the changes correctly.

### CODE REVIEW

File: src/OpenLoco/Audio/VehicleChannel.cpp [hunk 1]
```diff
@@ -1,18 +1,94 @@
...
+{
+    switch (zoom)
+    {
+        case 0:
+            return 0;
+        case 1:
+            return 35;
+        default:
+            return 70;
+    }
```
Comment (ZehMatt) [inline src/OpenLoco/Audio/VehicleChannel.cpp:22]: You could do this without branching.
```suggestion
    return std::min(2, zoom) * 35;
```

File: src/OpenLoco/Audio/VehicleChannel.cpp [hunk 2]
```diff
@@ -1,18 +1,94 @@
...
+    auto* w = Ui::WindowManager::find(v->soundWindowType, v->soundWindowNumber);
+    auto* viewport = w->viewports[0];
+    const auto uiPoint = viewport->viewportToScreen({ v->sprite_left, v->sprite_top });
+
+    const auto zoomVolumeModifier = getZoomVolumeModifier(viewport->zoom);
+
+    const auto relativePositionX = (uiPoint.x << 16) / std::max(Ui::width(), 64);
+    const auto panX = (relativePositionX - (1 << 15)) / 16;
+    const auto relativePositionY = (uiPoint.y << 16) / std::max(Ui::height(), 64);
+    const auto panY = (relativePositionY - (1 << 15)) / 16;
```
Comment (ZehMatt) [inline src/OpenLoco/Audio/VehicleChannel.cpp:50]: Consider moving this into a function like ```std::pair<int32_t, int32_t> calculcatePan(...)```

File: src/OpenLoco/Audio/VehicleChannel.cpp [hunk 3]
```diff
@@ -1,18 +1,94 @@
...
+        {
+            yFalloffModifier = 0;
+        }
+        else
+        {
+            yFalloffModifier = std::min((3072 - yFalloff) / 4, 255);
+        }
+    }
+
+    const auto falloffVolumeModifier = std::min(xFalloffModifier, yFalloffModifier);
```
Comment (ZehMatt) [inline src/OpenLoco/Audio/VehicleChannel.cpp:84]: Also better to move this into a new function.

File: src/OpenLoco/Audio/VehicleChannel.cpp (hunk 2 above)
Comment (duncanspumpkin) [reply src/OpenLoco/Audio/VehicleChannel.cpp:71]: yeah i already had but for some reason net connection went down when i uploaded :(

File: src/OpenLoco/Audio/VehicleChannel.cpp [hunk 4]
```diff
@@ -1,18 +1,83 @@
...
+        return falloffModifier;
+    }
+
+    static int32_t calculatePan(const coord_t coord, const int32_t screenSize)
+    {
+        const auto relativePosition = (coord << 16) / std::max(screenSize, 64);
+        return (relativePosition - (1 << 15)) / 16;
+    }
+
+    static std::pair<SoundId, ChannelAttributes> getChannelAttributesFromVehicle(const Vehicles::Vehicle2or6* v)
```
Comment (AaronVanGeffen) [inline src/OpenLoco/Audio/VehicleChannel.cpp:57]: Could you add some address comments?
```suggestion
    // 0x0048A590
    static std::pair<SoundId, ChannelAttributes> getChannelAttributesFromVehicle(const Vehicles::Vehicle2or6* v)
```

Comment (duncanspumpkin) [reply src/OpenLoco/Audio/VehicleChannel.cpp:57]: Ahh think that got nuked during my rebase.

File: src/OpenLoco/Audio/VehicleChannel.cpp [hunk 5]
```diff
@@ -1,18 +1,84 @@
...
 namespace OpenLoco::Audio
 {
-    static std::pair<SoundId, Channel::Attributes> sub_48A590(const Vehicles::Vehicle2or6* v)
+    static uint8_t getZoomVolumeModifier(uint8_t zoom)
     {
-        registers regs;
-        regs.esi = X86Pointer(v);
-        call(0x0048A590, regs);
-        return { static_cast<SoundId>(regs.eax), { regs.ecx, regs.edx, regs.ebx } };
+        return std::min<uint8_t>(zoom, 2) * 35;
```
Comment (LeftofZen) [inline src/OpenLoco/Audio/VehicleChannel.cpp:14]: So many magic numbers here, should we put them all into an `AudioConstants` file/class?

Comment (duncanspumpkin) [reply src/OpenLoco/Audio/VehicleChannel.cpp:14]: I think most would be specific to this file so could just have them at the top of the file

**Review by LeftofZen — CHANGES_REQUESTED** [review]
please reduce the amount of magic numbers in this file

File: src/OpenLoco/Audio/VehicleChannel.cpp [hunk 6]
```diff
@@ -1,18 +1,93 @@
...
+        const auto panY = calculatePan(uiPoint.y, Ui::height());
+
+        const auto undergroundVolumeModifier = getUndergroundVolumeModifier(v->position);
+
+        const auto xFalloffModifier = getFalloffModifier(panX);
+        const auto yFalloffModifier = getFalloffModifier(panY);
+
+        const auto falloffVolumeModifier = std::min(xFalloffModifier, yFalloffModifier);
+
+        const auto overalVolumeModifier = std::max(falloffVolumeModifier + undergroundVolumeModifier + zoomVolumeModifier, 0);
```
Comment (LeftofZen) [inline src/OpenLoco/Audio/VehicleChannel.cpp:85]: typo in the constant name, `overal` has only 1 'l'

File: src/OpenLoco/Audio/VehicleChannel.cpp [hunk 7]
```diff
@@ -1,18 +1,93 @@
...
+        return 0;
+    }
+
+    static uint8_t getFalloffModifier(int32_t pan)
     {
-        registers regs;
-        regs.esi = X86Pointer(v);
-        call(0x0048A590, regs);
-        return { static_cast<SoundId>(regs.eax), { regs.ecx, regs.edx, regs.ebx } };
+        const auto absPan = std::min(std::abs(pan), 4095); // This is pointless subsequent if's take care of max
```
Comment (LeftofZen) [inline src/OpenLoco/Audio/VehicleChannel.cpp:48]: what does 4095 represent, can it be a constant too?

File: src/OpenLoco/Audio/VehicleChannel.cpp [hunk 8]
```diff
@@ -1,18 +1,93 @@
...
 
 using namespace OpenLoco::Interop;
 
 namespace OpenLoco::Audio
 {
-    static std::pair<SoundId, Channel::Attributes> sub_48A590(const Vehicles::Vehicle2or6* v)
+    // Vehicle Volume is multiplied by the modifier then divided by 8 to get in terms of hundredth decibels
+    // 255 represents full volume
+    // 0 represents no volume
+    constexpr int8_t kVolumeModifierZoomIncrement = -35; // 13.7% decrease in volume for each zoom increment (up to 2)
```
Comment (LeftofZen) [inline src/OpenLoco/Audio/VehicleChannel.cpp:15]: these are nice, thanks for adding them

File: src/OpenLoco/Audio/VehicleChannel.cpp [hunk 9]
```diff
@@ -1,18 +1,93 @@
...
+    constexpr int8_t kVolumeModifierUnderground = -28;   // 11.0% decrease in volume when underground
+    constexpr uint8_t kVolumeModifierMax = 255;
+
+    constexpr int32_t kVolumeMin = -100'00; // hundredth decibels (-100dB)
+    // Calculated min is 255*255/8=8128 but original has done 256*256/8=8192
+    // We have kept original value but could be changed to correctly represent the full range of volume
+    constexpr int32_t kVehicleVolumeCalcMin = -81'91; // hundredth decibels (-81.91dB)
+
+    constexpr int32_t kPanFalloffStart = 2048;
+    constexpr int32_t kPanFalloffEnd = 3072;
```
Comment (LeftofZen) [inline src/OpenLoco/Audio/VehicleChannel.cpp:25]: these are nice, thanks for adding them

File: src/OpenLoco/Audio/VehicleChannel.cpp (hunk 7 above)
Comment (duncanspumpkin) [reply src/OpenLoco/Audio/VehicleChannel.cpp:48]: well the problem is is that its a pointless constant it doesn't do anything... I might just delete it.

**Review by LeftofZen — APPROVED** [review]

**Pull request merged**